src/VirtoCommerce.CatalogModule.Core/Model/CatalogProduct.cs

```diff
@@ -1,260 +1,271 @@
+
+        public IList<Image> Images { get; set; }
+        #endregion
+
+        #region IHasAssets members
+        public IList<Asset> Assets { get; set; }
+        #endregion
+
+        #region ILinkSupport members
+        public IList<CategoryLink> Links { get; set; }
+        #endregion
+
+        public IList<Variation> Variations { get; set; }
+        /// <summary>
+        /// Each descendant type should override this property to use other object type for seo records
+        /// </summary>
+        public virtual string SeoObjectType { get; } = "CatalogProduct";
+        public IList<SeoInfo> SeoInfos { get; set; }
+        public IList<EditorialReview> Reviews { get; set; }
+
+        #region IHasAssociations members
+        public IList<ProductAssociation> Associations { get; set; }
+        #endregion
+
+        public IList<ProductAssociation> ReferencedAssociations { get; set; }
+
+        #region IHasOutlines members
+        public IList<Outline> Outlines { get; set; }
+        #endregion
+
+        #region IInheritable members
+        /// <summary>
+        /// System flag used to mark that object was inherited from other
+        /// </summary>
         public bool IsInherited { get; private set; }
-       
-        public virtual void TryInheritFrom(IEntity parent)
-        {
-            if (parent is IHasProperties hasProperties)
-            {
-                //Properties inheritance
-                foreach (var parentProperty in hasProperties.Properties ?? Array.Empty<Property>())
-                {
-                    if (Properties == null)
-                    {
-                        Properties = new List<Property>();
-                    }
-                    var existProperty = Properties.FirstOrDefault(x => x.IsSame(parentProperty, PropertyType.Product, PropertyType.Variation));
-                    if (existProperty == null)
-                    {
-                        existProperty = AbstractTypeFactory<Property>.TryCreateInstance();
-                        Properties.Add(existProperty);
-                    }
-                    existProperty.TryInheritFrom(parentProperty);
-
-                    existProperty.IsReadOnly = existProperty.Type != PropertyType.Variation && existProperty.Type != PropertyType.Product;
-                }
-                //Restore sorting order after changes
-                if (Properties != null)
-                {
-                    Properties = Properties.OrderBy(x => x.Name).ToList();
-                }
+
+        public virtual void TryInheritFrom(IEntity parent)
+        {
+            this.InheritExcludedProperties(parent as IHasExcludedProperties);
+
+            if (parent is IHasProperties hasProperties)
+            {
+                //Properties inheritance
+                foreach (var parentProperty in hasProperties.Properties ?? Array.Empty<Property>())
+                {
+                    if (this.HasPropertyExcluded(parentProperty.Name))

```

Comment:
I  think this line is redundant, all excluded properties have already been  removed after the `InheritExcludedProperties`  call